### ISSUE TITLE
tilted ply model

### ISSUE THREAD

**Issue by jiangjiayao** [openMVG/openMVG#1547]
Hello, 

I successfully ran the SfM procedure on some spherical images, but the ply model is tilted. Does anyone know why it't tilted? Or how to prevent it from happening in the scripts? Or how to correct it after SfM?

Thanks a lot in advance
JJ

![image](https://user-images.githubusercontent.com/19272362/59974686-c19d4300-95e1-11e9-844b-7fc373aa0dff.png)

**Comment by pmoulon** [comment]
Does your image are flat, taken on a tripod with a perfect orientation?
- if this is the case, we can add a post process to restore the average rotation vector to be UP.

**Comment by jiangjiayao** [comment]
Yes, they were all taken with the same tripod on a flat surface. What's more, I think the camera has a gyroscope that keeps the pictures upright. @pmoulon 

Is there such a function in openMVG to restore the average rotation vector UP? Or do I have to write a custom function myself?

Thanks
JJ

**Comment by jiangjiayao** [comment]
Has anyone else have this issue? Does anyone know how to solve it? Thanks.

**Comment by pmoulon** [comment]
You can do something like.

Compute the deviation:
- for each rotation -> Ri * (0, 1, 0)
- average those vector
- Compute the rotation associated with this vector and apply the transpose of it to the entire scene  https://github.com/openMVG/openMVG/blob/master/src/openMVG/sfm/sfm_data_transform.hpp#L24

**Comment by pmoulon** [comment]
@jiangjiayao Did you work toward a working solution?
- Or do you want us (OpenMVG) community to provide a binary that would reset the camera of your choice to be `UP`?

**Comment by jiangjiayao** [comment]
I haven't been able to solve it. It would be great if the community can provide a solution. Thanks a lot @pmoulon
